de_web_plugin.cpp

```diff
@@ -15686,11 +15709,19 @@ void DeRestPluginPrivate::delayedFastEnddeviceProbe(const deCONZ::NodeEvent *eve
                         {
                             dateCode = attr.toString();
                             dateCodeAvailable = attr.isAvailable(); // might become false after first read
+                            if (dateCode.isEmpty())
+                            {
+                                dateCodeAvailable = false;
+                            }
                         }
                         else if (attr.id() == 0x4000 && swBuildId.isEmpty())
                         {
                             swBuildId = attr.toString();
                             swBuildIdAvailable = attr.isAvailable(); // might become false after first read
+                            if (swBuildId.isEmpty())
+                            {
+                                swBuildIdAvailable = false;
+                            }
```

Comment:
Please remove this, the changes break reading Datecode and SwBuildId since initially they will be empty. Note that the code a few lines later already checks for unavailable attributes and empty strings once they were read from the device.

Reply:
Ha ?
But even if Datecode  is initialy empty, we are after `dateCode = attr.toString();` so if the value is still empty, it will be always empty, no ?

>Note that the code a few lines later already checks for unavailable attributes and empty strings once they were read from the device

I m seeing for unavailable  but not for empty, I think it s for that there is this kind of hack

```
        if (node->nodeDescriptor().manufacturerCode() == VENDOR_SAMJIN)
        {
            swBuildIdAvailable = false; // empty string
            dateCodeAvailable = false; // unsupported attribute
        }
        else if (modelId == QLatin1String("HG06323")) // LIDL Remote Control
        {
            swBuildIdAvailable = false; // unsupported attribute
            dateCodeAvailable = false; // empty string
        }
```

And If I rermove this code I need to use hack too, else the inclusion fail, and I think ithere is the same issue too for a plug with attribute available but empty.

Reply:
> I m seeing for unavailable but not for empty, I think it s for that there is this kind of hack

The following code already marks the attribute as unavailable if the attribute was queried but has an empty string.

```c++
// line 14975

if (!attr.isAvailable())
{
    unavailBasicAttr.push_back(attr.id());
}
else if (attr.lastRead() != static_cast<time_t>(-1) && attr.dataType() == deCONZ::ZclCharacterString && attr.toString().isEmpty())
{
    // e.g. some Trust devices return empty strings
    unavailBasicAttr.push_back(attr.id());
}
```

But now when I look at the code it isn't used properly :)
I've created a PR https://github.com/dresden-elektronik/deconz-rest-plugin/pull/5092 to make this generic.

Reply:
Nice, some days I haven't see the user, but yes can work, need to test, and can solve issue for a tuya plug too.

Reply:
Would be good, I testet the PR with a Philips switch which has SwBuildId and Samsung sensors which have empty SwBuildId and Date Code not available — worked so far.